app/assets/javascripts/components/common/article_viewer_legend.jsx

```diff
@@ -25,23 +24,28 @@ const ArticleViewerLegend = ({ article, users, colors, status, allUsers, failure
 
       setUserLinks(users.map((user, i) => {
         let res;
+        // The 'unhighlightedContributors' array keeps track of the userids of users whose contributions
+        // were not successfully highlighted in the article viewer.
+        const isUserHovered = unhighlightedContributors.find(x => x === user.userid);
         const userLink = UserUtils.userTalkUrl(user.name, article.language, article.project);
         const fullUserRecord = allUsers.find(_user => _user.username === user.name);
         const realName = fullUserRecord && fullUserRecord.real_name;
         if (status === 'loading') {
           res = <div key={`legend-${user.name}`} className={'user-legend'}><a href={userLink} title={realName} target="_blank">{user.name}</a></div>;
         } else if (user.activeRevision === true) {
           res = <div key={`legend-${user.name}`} className={`user-legend ${colors[i]}`}><a href={userLink} title={realName} target="_blank">{user.name}</a><img className="user-legend-hover" style={{ color: 'transparent' }} src="/assets/images/arrow.svg" alt="scroll to users revisions" width="30px" height="20px" onClick={() => Scroller.scrollTo(user.name, scrollBox)} /></div >;
+        } else if (isUserHovered) {
```

Comment:
This will be false while the fallback request is being made, so during that request, a user will have the no_highlighting message even if they actually have contributions and are about to get the contributionsNotHighlighted message. Ideally, instead of that, it would wait until the fallback request is completed before adding either message. I'm okay with cutting that corner if it's hard to implement, though. Shouldn't be too much of a problem for users.